src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java

```diff
@@ -621,13 +628,36 @@ private void addBackButton(ChestMenu menu, int slot, Player p, PlayerProfile pro
     private static ItemStack getDisplayItem(Player p, boolean isSlimefunRecipe, ItemStack item) {
         if (isSlimefunRecipe) {
             SlimefunItem slimefunItem = SlimefunItem.getByItem(item);
+            boolean hasPerm = Slimefun.hasPermission(p, slimefunItem, false);
```

Comment:
Also don't declare variables before they are needed.
You are wasting unnecessary computations here because the next if-statement may render this unnecessary, therefore wasting memory.